// @the-orange-alliance/api/lib/models/game-specifics/SkystoneAllianceDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Stone;
(function (Stone) {
    Stone[Stone["UNKNOWN"] = -1] = "UNKNOWN";
    Stone[Stone["NONE"] = 0] = "NONE";
    Stone[Stone["STONE"] = 1] = "STONE";
    Stone[Stone["SKYSTONE"] = 2] = "SKYSTONE";
})(Stone = exports.Stone || (exports.Stone = {}));
const StoneMapping = {
    [Stone.UNKNOWN]: -1,
    [Stone.NONE]: 0,
    [Stone.STONE]: 1,
    [Stone.SKYSTONE]: 2
};
class SkystoneAllianceDetails {
    constructor() {
        this._autoStone1 = Stone.UNKNOWN;
        this._autoStone2 = Stone.UNKNOWN;
        this._autoStone3 = Stone.UNKNOWN;
        this._autoStone4 = Stone.UNKNOWN;
        this._autoStone5 = Stone.UNKNOWN;
        this._autoStone6 = Stone.UNKNOWN;
        this._autoDeliveredSkystones = -1;
        this._autoDeliveredStones = -1;
        this._autoReturned = -1;
        this._firstReturnedIsSkystone = false;
        this._autoPlaced = -1;
        this._foundationRepositioned = false;
        this._teleDelivered = -1;
        this._teleReturned = -1;
        this._telePlaced = -1;
        this._robot1Nav = false;
        this._robot2Nav = false;
        this._robot1Parked = false;
        this._robot2Parked = false;
        this._robot1CapLevel = -1;
        this._robot2CapLevel = -1;
        this._foundationMoved = false;
        this._autoTransportPts = -1;
        this._autoPlacedPts = -1;
        this._repositionPts = -1;
        this._navPts = -1;
        this._teleTransportPts = -1;
        this._telePlacedPts = -1;
        this._towerBonus = -1;
        this._cappingBonus = -1;
        this._levelBonus = -1;
        this._endRobotsParked = -1;
        this._autoPts = -1;
        this._auto = -1;
        this._tele = -1;
        this._end = -1;
        this._penalty = -1;
    }
    toJSON() {
        return {
            auto_stone_1: this.stoneToInt(this.autoStone1),
            auto_stone_2: this.stoneToInt(this.autoStone1),
            auto_stone_3: this.stoneToInt(this.autoStone1),
            auto_stone_4: this.stoneToInt(this.autoStone1),
            auto_stone_5: this.stoneToInt(this.autoStone1),
            auto_stone_6: this.stoneToInt(this.autoStone1),
            auto_delivered_skystones: this.autoDeliveredSkystones,
            auto_delivered_stones: this.autoDeliveredStones,
            auto_returned: this.autoReturned,
            first_returned_is_skystone: this.firstReturnedIsSkystone,
            auto_placed: this.autoPlaced,
            foundation_repositioned: this.foundationRepositioned,
            tele_delivered: this.teleDelivered,
            tele_returned: this.teleReturned,
            tele_placed: this.telePlaced,
            robot_1: {
                nav: this.robot1Nav,
                parked: this.robot1Parked,
                cap_level: this.robot1CapLevel
            },
            robot_2: {
                nav: this.robot2Nav,
                parked: this.robot2Parked,
                cap_level: this.robot2CapLevel
            },
            foundation_moved: this.foundationMoved,
            auto_transport_points: this.autoTransportPts,
            auto_placed_points: this.autoPlacedPts,
            reposition_points: this.repositionPts,
            nav_points: this.navPts,
            tele_transport_points: this.teleTransportPts,
            tele_placed_points: this.telePlacedPts,
            tower_bonus: this.towerBonus,
            tower_capping_bonus: this.cappingBonus,
            tower_level_bonus: this.levelBonus,
            end_robots_parked: this.endRobotsParked,
            auto_points: this.autoPts,
            auto_total: this.auto,
            tele_total: this.tele,
            end_total: this.end,
            penalty_total: this.penalty
        };
    }
    fromJSON(json) {
        const alliance = new SkystoneAllianceDetails();
        alliance.autoStone1 = this.intToStone(json.auto_stone_1);
        alliance.autoStone2 = this.intToStone(json.auto_stone_2);
        alliance.autoStone3 = this.intToStone(json.auto_stone_3);
        alliance.autoStone4 = this.intToStone(json.auto_stone_4);
        alliance.autoStone5 = this.intToStone(json.auto_stone_5);
        alliance.autoStone6 = this.intToStone(json.auto_stone_6);
        alliance.autoDeliveredSkystones = json.auto_delivered_skystones;
        alliance.autoDeliveredStones = json.auto_delivered_stones;
        alliance.autoReturned = json.auto_returned;
        alliance.firstReturnedIsSkystone = json.first_returned_is_skystone;
        alliance.autoPlaced = json.auto_placed;
        alliance.foundationRepositioned = json.foundation_repositioned;
        alliance.teleDelivered = json.tele_delivered;
        alliance.teleReturned = json.tele_returned;
        alliance.telePlaced = json.tele_placed;
        alliance.robot1Nav = json.robot_1.nav;
        alliance.robot2Nav = json.robot_2.nav;
        alliance.robot1Parked = json.robot_1.parked;
        alliance.robot2Parked = json.robot_2.parked;
        alliance.robot1CapLevel = json.robot_1.cap_level;
        alliance.robot2CapLevel = json.robot_2.cap_level;
        alliance.foundationMoved = json.foundation_moved;
        alliance.autoTransportPts = json.auto_transport_points;
        alliance.autoPlacedPts = json.auto_placed_points;
        alliance.repositionPts = json.reposition_points;
        alliance.navPts = json.nav_points;
        alliance.teleTransportPts = json.tele_transport_points;
        alliance.telePlacedPts = json.tele_placed_points;
        alliance.towerBonus = json.tower_bonus;
        alliance.cappingBonus = json.tower_capping_bonus;
        alliance.levelBonus = json.tower_level_bonus;
        alliance.endRobotsParked = json.end_robots_parked;
        alliance.autoPts = json.auto_points;
        alliance.auto = json.auto_total;
        alliance.tele = json.tele_total;
        alliance.end = json.end_total;
        alliance.penalty = json.penalty_total;
        return alliance;
    }
    intToStone(num) {
        return num;
    }
    stoneToInt(stone) {
        return StoneMapping[stone];
    }
    get autoStone1() {
        return this._autoStone1;
    }
    set autoStone1(value) {
        this._autoStone1 = value;
    }
    get autoStone2() {
        return this._autoStone2;
    }
    set autoStone2(value) {
        this._autoStone2 = value;
    }
    get autoStone3() {
        return this._autoStone3;
    }
    set autoStone3(value) {
        this._autoStone3 = value;
    }
    get autoStone4() {
        return this._autoStone4;
    }
    set autoStone4(value) {
        this._autoStone4 = value;
    }
    get autoStone5() {
        return this._autoStone5;
    }
    set autoStone5(value) {
        this._autoStone5 = value;
    }
    get autoStone6() {
        return this._autoStone6;
    }
    set autoStone6(value) {
        this._autoStone6 = value;
    }
    get autoDeliveredSkystones() {
        return this._autoDeliveredSkystones;
    }
    set autoDeliveredSkystones(value) {
        this._autoDeliveredSkystones = value;
    }
    get autoDeliveredStones() {
        return this._autoDeliveredStones;
    }
    set autoDeliveredStones(value) {
        this._autoDeliveredStones = value;
    }
    get autoReturned() {
        return this._autoReturned;
    }
    set autoReturned(value) {
        this._autoReturned = value;
    }
    get firstReturnedIsSkystone() {
        return this._firstReturnedIsSkystone;
    }
    set firstReturnedIsSkystone(value) {
        this._firstReturnedIsSkystone = value;
    }
    get autoPlaced() {
        return this._autoPlaced;
    }
    set autoPlaced(value) {
        this._autoPlaced = value;
    }
    get foundationRepositioned() {
        return this._foundationRepositioned;
    }
    set foundationRepositioned(value) {
        this._foundationRepositioned = value;
    }
    get teleDelivered() {
        return this._teleDelivered;
    }
    set teleDelivered(value) {
        this._teleDelivered = value;
    }
    get teleReturned() {
        return this._teleReturned;
    }
    set teleReturned(value) {
        this._teleReturned = value;
    }
    get telePlaced() {
        return this._telePlaced;
    }
    set telePlaced(value) {
        this._telePlaced = value;
    }
    get robot1Nav() {
        return this._robot1Nav;
    }
    set robot1Nav(value) {
        this._robot1Nav = value;
    }
    get robot2Nav() {
        return this._robot2Nav;
    }
    set robot2Nav(value) {
        this._robot2Nav = value;
    }
    get robot1Parked() {
        return this._robot1Parked;
    }
    set robot1Parked(value) {
        this._robot1Parked = value;
    }
    get robot2Parked() {
        return this._robot2Parked;
    }
    set robot2Parked(value) {
        this._robot2Parked = value;
    }
    get robot1CapLevel() {
        return this._robot1CapLevel;
    }
    set robot1CapLevel(value) {
        this._robot1CapLevel = value;
    }
    get robot2CapLevel() {
        return this._robot2CapLevel;
    }
    set robot2CapLevel(value) {
        this._robot2CapLevel = value;
    }
    get foundationMoved() {
        return this._foundationMoved;
    }
    set foundationMoved(value) {
        this._foundationMoved = value;
    }
    get autoTransportPts() {
        return this._autoTransportPts;
    }
    set autoTransportPts(value) {
        this._autoTransportPts = value;
    }
    get autoPlacedPts() {
        return this._autoPlacedPts;
    }
    set autoPlacedPts(value) {
        this._autoPlacedPts = value;
    }
    get repositionPts() {
        return this._repositionPts;
    }
    set repositionPts(value) {
        this._repositionPts = value;
    }
    get navPts() {
        return this._navPts;
    }
    set navPts(value) {
        this._navPts = value;
    }
    get teleTransportPts() {
        return this._teleTransportPts;
    }
    set teleTransportPts(value) {
        this._teleTransportPts = value;
    }
    get telePlacedPts() {
        return this._telePlacedPts;
    }
    set telePlacedPts(value) {
        this._telePlacedPts = value;
    }
    get towerBonus() {
        return this._towerBonus;
    }
    set towerBonus(value) {
        this._towerBonus = value;
    }
    get cappingBonus() {
        return this._cappingBonus;
    }
    set cappingBonus(value) {
        this._cappingBonus = value;
    }
    get levelBonus() {
        return this._levelBonus;
    }
    set levelBonus(value) {
        this._levelBonus = value;
    }
    get endRobotsParked() {
        return this._endRobotsParked;
    }
    set endRobotsParked(value) {
        this._endRobotsParked = value;
    }
    get autoPts() {
        return this._autoPts;
    }
    set autoPts(value) {
        this._autoPts = value;
    }
    get auto() {
        return this._auto;
    }
    set auto(value) {
        this._auto = value;
    }
    get tele() {
        return this._tele;
    }
    set tele(value) {
        this._tele = value;
    }
    get end() {
        return this._end;
    }
    set end(value) {
        this._end = value;
    }
    get penalty() {
        return this._penalty;
    }
    set penalty(value) {
        this._penalty = value;
    }
}
exports.default = SkystoneAllianceDetails;
