// @the-orange-alliance/api/lib/models/EventLiveStream.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const StreamType_1 = require("./types/StreamType");
class EventLiveStream {
    constructor() {
        this._streamKey = "";
        this._eventKey = "";
        this._streamType = StreamType_1.default.Unknown;
        this._isActive = false;
        this._streamURL = "";
        this._channelName = "";
        this._streamName = "";
        this._startDateTime = "";
        this._endDateTime = "";
        this._channelURL = "";
        this._fullURL = "";
    }
    toJSON() {
        return {
            stream_key: this.streamKey,
            event_key: this.eventKey,
            stream_type: StreamType_1.streamToNumber(this.streamType),
            is_active: this.isActive,
            url: this.streamURL,
            channel_name: this.channelName,
            stream_name: this.streamName,
            start_datetime: this.startDateTime,
            end_datetime: this.endDateTime,
            channel_url: this.channelURL
        };
    }
    fromJSON(json) {
        const stream = new EventLiveStream();
        stream.streamKey = json.stream_key;
        stream.eventKey = json.event_key;
        stream.streamType = StreamType_1.numberToStream(json.stream_type);
        stream.isActive = json.is_active;
        stream.streamURL = json.url;
        stream.channelName = json.channel_name;
        stream.streamName = json.stream_name;
        stream.startDateTime = json.start_datetime;
        stream.endDateTime = json.end_datetime;
        stream.channelURL = json.channel_url;
        return stream;
    }
    get streamKey() {
        return this._streamKey;
    }
    set streamKey(value) {
        this._streamKey = value;
    }
    get eventKey() {
        return this._eventKey;
    }
    set eventKey(value) {
        this._eventKey = value;
    }
    get streamType() {
        return this._streamType;
    }
    set streamType(value) {
        this._streamType = value;
    }
    get isActive() {
        return this._isActive;
    }
    set isActive(value) {
        this._isActive = value;
    }
    get streamURL() {
        return this._streamURL;
    }
    set streamURL(value) {
        this._streamURL = value;
    }
    get channelName() {
        return this._channelName;
    }
    set channelName(value) {
        this._channelName = value;
    }
    get streamName() {
        return this._streamName;
    }
    set streamName(value) {
        this._streamName = value;
    }
    get startDateTime() {
        return this._startDateTime;
    }
    set startDateTime(value) {
        this._startDateTime = value;
    }
    get endDateTime() {
        return this._endDateTime;
    }
    set endDateTime(value) {
        this._endDateTime = value;
    }
    get channelURL() {
        return this._channelURL;
    }
    set channelURL(value) {
        this._channelURL = value;
    }
    get fullURL() {
        return this._fullURL;
    }
    set fullURL(value) {
        this._fullURL = value;
    }
}
exports.default = EventLiveStream;
