// @the-orange-alliance/api/lib/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Alliance_1 = require("./Alliance");
exports.Alliance = Alliance_1.default;
var Award_1 = require("./Award");
exports.Award = Award_1.default;
var AwardRecipient_1 = require("./AwardRecipient");
exports.AwardRecipient = AwardRecipient_1.default;
var Event_1 = require("./Event");
exports.Event = Event_1.default;
var EventLiveStream_1 = require("./EventLiveStream");
exports.EventLiveStream = EventLiveStream_1.default;
var EventParticipant_1 = require("./EventParticipant");
exports.EventParticipant = EventParticipant_1.default;
var Insights_1 = require("./Insights");
exports.Insights = Insights_1.default;
var League_1 = require("./League");
exports.League = League_1.default;
var LeagueDiv_1 = require("./LeagueDiv");
exports.LeagueDiv = LeagueDiv_1.default;
var Match_1 = require("./Match");
exports.Match = Match_1.default;
var MatchDetails_1 = require("./MatchDetails");
exports.MatchDetails = MatchDetails_1.default;
var MatchParticipant_1 = require("./MatchParticipant");
exports.MatchParticipant = MatchParticipant_1.default;
var Media_1 = require("./Media");
exports.Media = Media_1.default;
var ModifiedEvent_1 = require("./ModifiedEvent");
exports.ModifiedEvent = ModifiedEvent_1.default;
var ModifiedTeam_1 = require("./ModifiedTeam");
exports.ModifiedTeam = ModifiedTeam_1.default;
var Ranking_1 = require("./Ranking");
exports.Ranking = Ranking_1.default;
var Region_1 = require("./Region");
exports.Region = Region_1.default;
var Season_1 = require("./Season");
exports.Season = Season_1.default;
var Team_1 = require("./Team");
exports.Team = Team_1.default;
var TeamSeasonRecord_1 = require("./TeamSeasonRecord");
exports.TeamSeasonRecord = TeamSeasonRecord_1.default;
var WebAnnouncement_1 = require("./WebAnnouncement");
exports.WebAnnouncement = WebAnnouncement_1.default;
