// @the-orange-alliance/api/lib/models/MatchParticipant.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const StationStatus_1 = require("./types/StationStatus");
const MatchStations_1 = require("./types/MatchStations");
const RefStatus_1 = require("./types/RefStatus");
class MatchParticipant {
    constructor() {
        this._matchParticipantKey = "";
        this._matchKey = "";
        this._teamKey = "";
        this._teamNumber = -1;
        this._station = MatchStations_1.default.Unknown;
        this._stationStatus = StationStatus_1.default.Unknown;
        this._refStatus = 0;
    }
    toJSON() {
        return {
            match_participant_key: this.matchParticipantKey,
            match_key: this.matchKey,
            team_key: this.teamKey,
            team_number: this.teamNumber,
            station: MatchStations_1.stationToNumber(this.station),
            station_status: StationStatus_1.stationStatusToNumber(this.stationStatus),
            ref_status: RefStatus_1.refToNumber(this.refStatus)
        };
    }
    fromJSON(json) {
        const participant = new MatchParticipant();
        participant.matchParticipantKey = json.match_participant_key;
        participant.matchKey = json.match_key;
        participant.teamKey = json.team_key;
        participant.teamNumber = json.team_number;
        participant.station = MatchStations_1.numberToStation(json.station);
        participant.stationStatus = StationStatus_1.numberToStationStatus(json.station_status);
        participant.refStatus = RefStatus_1.numberToRef(json.ref_status);
        return participant;
    }
    get matchParticipantKey() {
        return this._matchParticipantKey;
    }
    set matchParticipantKey(value) {
        this._matchParticipantKey = value;
    }
    get matchKey() {
        return this._matchKey;
    }
    set matchKey(value) {
        this._matchKey = value;
    }
    get teamKey() {
        return this._teamKey;
    }
    set teamKey(value) {
        this._teamKey = value;
    }
    get teamNumber() {
        return this._teamNumber;
    }
    set teamNumber(value) {
        this._teamNumber = value;
    }
    get station() {
        return this._station;
    }
    set station(value) {
        this._station = value;
    }
    get stationStatus() {
        return this._stationStatus;
    }
    set stationStatus(value) {
        this._stationStatus = value;
    }
    get refStatus() {
        return this._refStatus;
    }
    set refStatus(value) {
        this._refStatus = value;
    }
}
exports.default = MatchParticipant;
