// @the-orange-alliance/api/lib/models/Media.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const MediaType_1 = require("./types/MediaType");
class Media {
    constructor() {
        this._mediaKey = "";
        this._eventKey = "";
        this._teamKey = "";
        this._mediaType = MediaType_1.MediaTypeTeam.Unknown;
        this._isPrimary = false;
        this._mediaTitle = "";
        this._mediaLink = "";
    }
    toJSON() {
        let m_type = MediaType_1.MediaTypeTeam.Unknown;
        if (this.teamKey !== "" || this.teamKey !== undefined) {
            m_type = MediaType_1.teamMediaToNumber(this.mediaType);
        }
        else if (this.eventKey !== "" || this.eventKey !== undefined) {
            m_type = MediaType_1.eventMediaToNumber(this.mediaType);
        }
        else {
            m_type = -1;
        }
        return {
            media_key: this.mediaKey,
            event_key: this.eventKey,
            team_key: this.teamKey,
            media_type: m_type,
            is_primary: this.isPrimary,
            title: this.mediaTitle,
            link: this.mediaLink
        };
    }
    fromJSON(json) {
        const award = new Media();
        if (this.teamKey !== "") {
            award.mediaType = MediaType_1.numberToTeamMedia(json.media_type);
        }
        else if (this.teamKey !== "") {
            award.mediaType = MediaType_1.numberToEventMedia(json.media_type);
        }
        else {
            award.mediaType = MediaType_1.MediaTypeTeam.Unknown;
        }
        award.mediaKey = json.media_key;
        award.eventKey = json.event_key;
        award.teamKey = json.team_key;
        award.isPrimary = json.is_primary;
        award.mediaTitle = json.title;
        award.mediaLink = json.link;
        return award;
    }
    get mediaKey() {
        return this._mediaKey;
    }
    set mediaKey(value) {
        this._mediaKey = value;
    }
    get eventKey() {
        return this._eventKey;
    }
    set eventKey(value) {
        this._eventKey = value;
    }
    get teamKey() {
        return this._teamKey;
    }
    set teamKey(value) {
        this._teamKey = value;
    }
    get mediaType() {
        return this._mediaType;
    }
    set mediaType(value) {
        this._mediaType = value;
    }
    get isPrimary() {
        return this._isPrimary;
    }
    set isPrimary(value) {
        this._isPrimary = value;
    }
    get mediaTitle() {
        return this._mediaTitle;
    }
    set mediaTitle(value) {
        this._mediaTitle = value;
    }
    get mediaLink() {
        return this._mediaLink;
    }
    set mediaLink(value) {
        this._mediaLink = value;
    }
}
exports.default = Media;
