// @the-orange-alliance/api/lib/models/types/MatchType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var MatchType;
(function (MatchType) {
    MatchType[MatchType["Unknown"] = -1] = "Unknown";
    MatchType[MatchType["PracticeMatch"] = 0] = "PracticeMatch";
    MatchType[MatchType["QualificationMatch"] = 1] = "QualificationMatch";
    MatchType[MatchType["QuarterfinalSeries1"] = 21] = "QuarterfinalSeries1";
    MatchType[MatchType["QuarterfinalSeries2"] = 22] = "QuarterfinalSeries2";
    MatchType[MatchType["QuarterfinalSeries3"] = 23] = "QuarterfinalSeries3";
    MatchType[MatchType["QuarterfinalSeries4"] = 24] = "QuarterfinalSeries4";
    MatchType[MatchType["SemifinalsSeries1"] = 31] = "SemifinalsSeries1";
    MatchType[MatchType["SemifinalsSeries2"] = 32] = "SemifinalsSeries2";
    MatchType[MatchType["Finals"] = 4] = "Finals";
})(MatchType || (MatchType = {}));
const MatchTypeMapping = {
    [MatchType.Unknown]: -1,
    [MatchType.PracticeMatch]: 0,
    [MatchType.QualificationMatch]: 1,
    [MatchType.QuarterfinalSeries1]: 21,
    [MatchType.QuarterfinalSeries2]: 22,
    [MatchType.QuarterfinalSeries3]: 23,
    [MatchType.QuarterfinalSeries4]: 24,
    [MatchType.SemifinalsSeries1]: 31,
    [MatchType.SemifinalsSeries2]: 32,
    [MatchType.Finals]: 4
};
exports.numberToMatchType = (matchType) => {
    return matchType;
};
exports.matchTypeToNumber = (matchType) => {
    return MatchTypeMapping[matchType];
};
exports.default = MatchType;
