// @the-orange-alliance/api/lib/models/types/MediaType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var MediaTypeTeam;
(function (MediaTypeTeam) {
    MediaTypeTeam[MediaTypeTeam["Unknown"] = -1] = "Unknown";
    MediaTypeTeam[MediaTypeTeam["Github"] = 0] = "Github";
    MediaTypeTeam[MediaTypeTeam["CAD"] = 1] = "CAD";
    MediaTypeTeam[MediaTypeTeam["EngNotebook"] = 2] = "EngNotebook";
    MediaTypeTeam[MediaTypeTeam["RobotReveal"] = 3] = "RobotReveal";
    MediaTypeTeam[MediaTypeTeam["RobotImage"] = 4] = "RobotImage";
    MediaTypeTeam[MediaTypeTeam["TeamLogo"] = 5] = "TeamLogo";
})(MediaTypeTeam || (MediaTypeTeam = {}));
exports.MediaTypeTeam = MediaTypeTeam;
const TeamMediaMapping = {
    [MediaTypeTeam.Unknown]: -1,
    [MediaTypeTeam.Github]: 0,
    [MediaTypeTeam.CAD]: 1,
    [MediaTypeTeam.EngNotebook]: 2,
    [MediaTypeTeam.RobotReveal]: 3,
    [MediaTypeTeam.RobotImage]: 4,
    [MediaTypeTeam.TeamLogo]: 5
};
const numberToTeamMedia = (mediaType) => {
    return mediaType;
};
exports.numberToTeamMedia = numberToTeamMedia;
const teamMediaToNumber = (media) => {
    return TeamMediaMapping[media];
};
exports.teamMediaToNumber = teamMediaToNumber;
var MediaTypeEvent;
(function (MediaTypeEvent) {
    MediaTypeEvent[MediaTypeEvent["Unknown"] = -1] = "Unknown";
    MediaTypeEvent[MediaTypeEvent["PitMap"] = 0] = "PitMap";
    MediaTypeEvent[MediaTypeEvent["Schedule"] = 1] = "Schedule";
    MediaTypeEvent[MediaTypeEvent["VenueMap"] = 2] = "VenueMap";
    MediaTypeEvent[MediaTypeEvent["EventPhoto"] = 6] = "EventPhoto";
})(MediaTypeEvent || (MediaTypeEvent = {}));
exports.MediaTypeEvent = MediaTypeEvent;
const EventMediaMapping = {
    [MediaTypeEvent.Unknown]: -1,
    [MediaTypeEvent.PitMap]: 0,
    [MediaTypeEvent.Schedule]: 1,
    [MediaTypeEvent.VenueMap]: 2,
    [MediaTypeEvent.EventPhoto]: 6
};
const numberToEventMedia = (mediaType) => {
    return mediaType;
};
exports.numberToEventMedia = numberToEventMedia;
const eventMediaToNumber = (media) => {
    return EventMediaMapping[media];
};
exports.eventMediaToNumber = eventMediaToNumber;
